nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -450,10 +483,186 @@ impl DataStore {
                         .await?;
                 }
 
+                // Insert pending MGS updates for service processors for this
+                // blueprint.  These include foreign keys into the
+                // hw_baseboard_id table that we don't have handy.  To achieve
+                // this, we use the same pattern used during inventory
+                // insertion:
+                //
+                //   INSERT INTO bp_pending_mgs_update_sp
+                //       SELECT
+                //           id
+                //           [other column values as literals]
+                //         FROM hw_baseboard_id
+                //         WHERE part_number = ... AND serial_number = ...;
+                //
+                // This way, we don't need to know the id.  The database looks
+                // it up for us as it does the INSERT.
+
+                for update in &blueprint.pending_mgs_updates {
+                    // Right now, we only implement support for storing SP
+                    // updates.
+                    let (expected_active_version, expected_inactive_version) =
+                        match &update.details {
+                            PendingMgsUpdateDetails::Sp {
+                                expected_active_version,
+                                expected_inactive_version,
+                            } => (
+                                expected_active_version,
+                                expected_inactive_version,
+                            ),
+                            PendingMgsUpdateDetails::Rot { .. }
+                            | PendingMgsUpdateDetails::RotBootloader {
+                                ..
+                            } => continue,
+                        };
```

Comment:
How will this fit with the RoT and RoT bootloader? Specifically with the RoT, will we only want the active/inactive versions like it's represented here? Or will we want to have all of the other additional fields (active slot, persistent boot preference, etc)?

If it's the latter, is the idea to have additional columns for these (`active_slot`, `slot_a_version` etc for RoT, and `stage0_version`, `stage0_next_version` for bootloader) and leave `expected_active_version` and `expected_inactive_version` empty? Or do we want completely different tables for these two (`bp_pending_mgs_update_rot`, `bp_pending_mgs_update_rot_bootloader`)?